cli-tests/src/utils.rs

```diff
@@ -33,49 +33,99 @@ pub fn generate_mock_git_repo<T: AsRef<Path>>(directory: T) {
-            payload.test_action_output = vec![File {
+        .zip(reports.iter())
+        .flat_map(|(junit, report)| {
+            let file = File {
                 name: junit.file_name().unwrap().to_str().unwrap().to_string(),
                 file: Some(Uri(junit.to_string_lossy().to_string())),
                 ..Default::default()
-            }];
-            build_event.payload = Some(Payload::TestResult(payload));
-            build_event.id = Some(BuildEventId {
-                id: Some(Id::TestResult(TestResultId {
-                    label: "//path:test".to_string(),
-                    ..Default::default()
-                })),
-            });
-            build_event
+            };
+            let test_start_time = report.timestamp.map(|ts| ts.to_utc().into());
+            let test_duration = report.time.map(|d| d.into());
+            [
+                BuildEvent {
+                    id: Some(BuildEventId {
+                        id: Some(Id::TestResult(TestResultId {
+                            label: "//path:test".to_string(),
+                            ..Default::default()
+                        })),
+                    }),
+                    children: vec![],
+                    last_message: false,
+                    payload: Some(Payload::TestResult(TestResult {
+                        status: TestStatus::Passed.into(),
+                        test_attempt_start: test_start_time.clone(),
+                        test_attempt_duration: test_duration.clone(),
+                        test_action_output: vec![file.clone()],
+                        ..Default::default()
+                    })),
+                },
+                BuildEvent {
+                    id: Some(BuildEventId {
+                        id: Some(Id::TestSummary(TestSummaryId {
+                            label: "//path:test".to_string(),
+                            ..Default::default()
+                        })),
+                    }),
+                    children: vec![],
+                    last_message: false,
+                    payload: Some(Payload::TestSummary(TestSummary {
+                        overall_status: TestStatus::Passed.into(),
+                        total_run_count: 1,
+                        run_count: 1,
+                        attempt_count: 1,
+                        shard_count: 1,
+                        passed: vec![file],
+                        failed: vec![],
+                        total_num_cached: 0,
+                        first_start_time: test_start_time.clone(),
+                        last_stop_time: Some(
+                            (report.timestamp.unwrap() + report.time.unwrap())
+                                .to_utc()
+                                .into(),
+                        ),
+                        total_run_duration: test_duration,
+                        ..Default::default()
+                    })),
```

Comment:
Every BEP I've inspected includes the summary which we use to set the resolved status for tests. Additionally, this will be used for the start and end times in follow up PRs.